src/worldcereal/utils/refdata.py

```diff
@@ -132,12 +136,116 @@ def query_public_extractions(
     public_df_raw = db.sql(main_query).df()
 
     # Process the parquet into the format we need for training
-    processed_public_df = process_parquet(public_df_raw)
+    processed_public_df = process_parquet(public_df_raw, processing_period)
 
     return processed_public_df
 
 
-def process_parquet(public_df_raw: pd.DataFrame) -> pd.DataFrame:
+def month_diff(month1: int, month2: int) -> int:
+    """
+    Calculate the number of months between two given arbitrary months.
+    This function computes the difference in months between `month1` and `month2`
+    assuming that `month2` is in the past relative to `month1`.
+    The difference is calculated such that it falls within the range of 0 to 12 months.
+    Args:
+        month1 (int): The reference month (1-12).
+        month2 (int): The month to compare against (1-12).
+    Returns:
+        int: The number of months between `month1` and `month2`.
+    """
+
+    dummy_year = 2020
+    dummy_date1 = pd.to_datetime(f"{dummy_year}-{month1}-01")
+
+    date2_candidate1 = pd.to_datetime(f"{dummy_year}-{month2}-01")
+    date2_candidate2 = pd.to_datetime(f"{dummy_year-1}-{month2}-01")
+
+    shift1 = (dummy_date1.year * 12 + dummy_date1.month) - (
+        date2_candidate1.year * 12 + date2_candidate1.month
+    )
+
+    shift2 = (dummy_date1.year * 12 + dummy_date1.month) - (
+        date2_candidate2.year * 12 + date2_candidate2.month
+    )
+
+    shift = [xx for xx in [shift1, shift2] if (xx > 0) and (xx <= 12)][0]
+    if shift == 12:
+        shift = 0
+
+    return shift
+
+
+def get_best_valid_date(row):
+    """
+    Determine the best valid date for a given row based on forward and backward shifts.
```

Comment:
is this a standard Numpy docstring? Hard to see from github changes, just checking with you.